lib/api/src/engine.rs

```diff
@@ -90,11 +99,43 @@ impl Engine {
         &self,
         file_ref: &Path,
     ) -> Result<Arc<Artifact>, DeserializeError> {
-        let mut file = std::fs::File::open(file_ref)?;
-        let mut buffer = Vec::new();
-        // read the whole file
-        file.read_to_end(&mut buffer)?;
-        Ok(Arc::new(Artifact::deserialize(&self.0, buffer.as_slice())?))
+        let bytes = std::fs::read(file_ref)?;
+        Ok(Arc::new(Artifact::deserialize(&self.0, bytes.into())?))
+    }
+
+    #[cfg(all(feature = "sys", not(target_arch = "wasm32")))]
+    /// Load a serialized WebAssembly module from a memory mapped file and deserialize it.
+    ///
+    /// NOTE: you should almost always prefer [`Self::deserialize_from_mmapped_file`].
+    ///
+    /// # Safety
+    /// See [`Artifact::deserialize_unchecked`].
+    pub unsafe fn deserialize_from_mmapped_file_unchecked(
```

Comment:
We started to move architecture specific implementations to `NativeEngineExt` instead of inherent impls.

Since mmap isn't available for WASM targets, let's move the `mmaped_*` methods to that trait.

Reply:
All of the deserialize methods are `#[cfg(all(feature = "sys", not(target_arch = "wasm32")))]`. Do we want to keep the other ones here and just move the mmapped functions?

Reply:
We should probably have a followup task though to move them whenever `5.0` comes around.

Reply:
Done. I also created #4143.